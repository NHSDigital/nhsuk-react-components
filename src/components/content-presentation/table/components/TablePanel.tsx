import React, { FC, ComponentProps, HTMLProps } from 'react';
import classNames from 'classnames';
import HeadingLevel from '@components/utils/HeadingLevel';

export interface TablePanelProps extends HTMLProps<HTMLDivElement> {
  heading?: string;
  headingProps?: ComponentProps<typeof HeadingLevel>;
}

const TablePanel: FC<TablePanelProps> = ({
  className,
  heading,
  headingProps,
  children,
  ...rest
}) => (
  <div className={classNames('nhsuk-table__panel-with-heading-tab', className)} {...rest}>
    {heading && (
      <HeadingLevel
        headingLevel="h3"
        {...headingProps}
        className={classNames('nhsuk-table__heading-tab', headingProps?.className)}
      >
        {heading}
      </HeadingLevel>
    )}
    {children}
  </div>
);

export default TablePanel;
