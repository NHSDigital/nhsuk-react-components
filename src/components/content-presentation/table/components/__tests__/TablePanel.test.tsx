import React from 'react';
import { render } from '@testing-library/react';
import TablePanel from '../TablePanel';

describe('Table.Panel', () => {
  it('matches snapshot', () => {
    const { container } = render(<TablePanel />);

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when rendering a h2 heading', () => {
    const { container } = render(
      <TablePanel heading="TestHeading" headingProps={{ headingLevel: 'h2' }} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('adds header when prop added', () => {
    const { container } = render(
      <TablePanel heading="TestHeading" headingProps={{ headingLevel: 'h2' }} />,
    );
    const heading = container.querySelector('h2.nhsuk-table__heading-tab');

    expect(heading).toBeTruthy();
  });
});
