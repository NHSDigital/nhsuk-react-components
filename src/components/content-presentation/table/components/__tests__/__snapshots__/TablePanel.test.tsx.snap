// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.Panel matches snapshot 1`] = `
<div>
  <div
    class="nhsuk-table__panel-with-heading-tab"
  />
</div>
`;

exports[`Table.Panel matches snapshot when rendering a h2 heading 1`] = `
<div>
  <div
    class="nhsuk-table__panel-with-heading-tab"
  >
    <h2
      class="nhsuk-table__heading-tab"
    >
      TestHeading
    </h2>
  </div>
</div>
`;
